cpp/src/arrow/compute/api_aggregate.h

```diff
@@ -175,6 +175,88 @@ class ARROW_EXPORT TDigestOptions : public FunctionOptions {
   uint32_t min_count;
 };
 
+/// \brief Control Pivot kernel behavior
+///
+/// These options apply to the "pivot_wider" and "hash_pivot_wider" functions.
+///
+/// Constraints:
+/// - The corresponding `Aggregate::target` must have two FieldRef elements;
```

Comment:
Do we not support multiple "key" or "data" columns? Or is the user expected to use a `struct` to wrap the desired key/data columns to achieve that?

Reply:
Multiple pivot key columns are not supported at all (the pivot key must be string-like). Multiple pivot value columns can indeed be achieved using a `struct`.

Reply:
But, of course, you can pass several aggregates to a group_by, so you can have several hash_pivot's at once if you want.

Reply:
Good point. Thank you.